releasenotes-builder/src/main/java/com/github/checkstyle/Main.java

```diff
@@ -69,6 +69,7 @@ public static void main(String... args) {
                     final List<String> validationErrors = MainProcess
                         .validateNotes(notesBuilderResult.getReleaseNotes(), cliOptions);
                     notesBuilderResult.addErrors(validationErrors);
+                    printListOf(validationErrors);
```

Comment:
`runGithubNotesBuilder` has similar code on line 122 and 125.

Why don't we just remove that code, and move it to the end of this method so the last thing we do before ending is display all warnings and errors? This is what we seem to do with `publicationErrors`.

Reply:
Yes, this becomes tricky.
We have violations that we need to print before release notes validation and printing notes. As we can not do release number validation if label is not set.

We need to print results of release notes publishing.

At the very end we need to do validation of release number and print violation on it. Violation on release number should be after print of release notes to let maintainer see what should be fixed - release number or change labels.

@stoyanK7 , can we archive above sequence?
I am ok with more refactoring or keeping two different set of errors in result object.